packages/apollo-gateway/src/index.ts

```diff
@@ -108,25 +162,85 @@ export class ApolloGateway implements GraphQLService {
+        config.experimental_didResolveQueryPlan;
+      this.experimental_didFailComposition =
+        config.experimental_didFailComposition;
+      this.experimental_didUpdateComposition =
+        config.experimental_didUpdateComposition;
+      this.experimental_pollInterval = config.experimental_pollInterval;
+
+      // ie if they have a pollinterval and a custom loader or a serviceList
+      if (config.experimental_pollInterval) {
+        if (
+          config.experimental_updateServiceDefinitions ||
+          (config as RemoteGatewayConfig).serviceList
+        ) {
+          console.warn(
+            'Polling running services is dangerous and not recommended in production. ' +
+              'Polling should only be used against a registry. ' +
+              'If you are polling running services, use with caution.',
+          );
+        }
+      }
+    }
   }
 
   public async load(options?: { engine?: GraphQLServiceEngineConfig }) {
-    if (options && options.engine) {
-      if (!options.engine.graphVariant)
-        console.warn('No graph variant provided. Defaulting to `current`.');
-      this.engineConfig = options.engine;
-    }
+    const loader = async () => {
+      const previousSchema = this.schema;
+      const previousServiceDefinitions = this.serviceDefinitions;
+      if (options && options.engine) {
+        if (!options.engine.graphVariant)
+          console.warn('No graph variant provided. Defaulting to `current`.');
+        this.engineConfig = options.engine;
+      }
 
-    if (this.schema) {
-      return { schema: this.schema, executor: this.executor };
-    }
+      if (this.schema) {
+        return { schema: this.schema, executor: this.executor };
+      }
+
+      this.logger.debug('Loading configuration for Gateway');
+      const [services] = await this.updateServiceDefinitions(this.config);
+
+      this.logger.debug('Configuration loaded for Gateway');
+
+      this.schema = this.createSchema(services);
+      this.serviceDefinitions = services;
+
+      if (this.experimental_didUpdateComposition) {
+        this.experimental_didUpdateComposition(
+          {
+            serviceDefinitions: services,
+            schema: this.schema,
+          },
+          previousServiceDefinitions &&
+            previousSchema && {
+              serviceDefinitions: previousServiceDefinitions,
+              schema: previousSchema,
+            },
+        );
+      }
+    };
 
-    this.logger.debug('Loading configuration for Gateway');
-    const [services] = await this.loadServiceDefinitions(this.config);
-    this.logger.debug('Configuration loaded for Gateway');
-    this.schema = this.createSchema(services);
+    await loader();
 
-    return { schema: this.schema, executor: this.executor };
+    return {
+      schema: this.schema,
+      executor: this.executor,
+      interval: this.experimental_pollInterval
+        ? setInterval(loader, this.experimental_pollInterval)
```

Comment:
isn't `setInterval` some kind of async timer thing?

Reply:
Yep! should allow us to run what was the `load` function on an interval. I run the function once manually above this to make sure it's run if there isn't a pollInterval passed.

Reply:
I'm returning it, so it can be cancelled by the caller–the tests for my case.

Reply:
Also, thinking back to the comments on the interval setting and not recommending it with the default poller, why don't we literally throw an error in that case rather than letting people do a not recommended thing?

Reply:
@zionts I think we wanted to let implementers be able to do anything they _needed_ to do including for strange use-cases, but let them know this is dangerous. 

Maybe we could add an option to silence this warning in the future once not `experimental_`?

Reply:
Also, we want to warn this use case, but there's no way to _know_ they're not polling a running service if they implement their own fetcher.